File: src/core/jupiter/core/prm/sub/person/sub/occasion/kind.ts
import { OccasionKind } from "@jupiter/webapi-client";

export function occasionKindName(kind: OccasionKind): string {
  switch (kind) {
    case OccasionKind.BIRTHDAY:
      return "Birthday";
    case OccasionKind.ANNIVERSARY:
      return "Anniversary";
    case OccasionKind.HOLIDAY:
      return "Holiday";
    case OccasionKind.OTHER:
      return "Other";
  }
}

File: src/core/jupiter/core/prm/sub/person/sub/occasion/kind.py
"""The kind of an occasion."""

from jupiter.framework.value import EnumValue, enum_value


@enum_value
class OccasionKind(EnumValue):
    """The kind of an occasion."""

    BIRTHDAY = "birthday"
    ANNIVERSARY = "anniversary"
    HOLIDAY = "holiday"
    OTHER = "other"
